source/common/router/retry_state_impl.cc

```diff
@@ -216,6 +226,14 @@ bool RetryStateImpl::wouldRetry(const Http::HeaderMap* response_headers,
     }
   }
 
+  if ((retry_on_ & RetryPolicy::RETRY_ON_RETRIABLE_STATUS_CODES)) {
+    for (auto code : retriable_status_codes_) {
+      if (Http::Utility::getResponseStatus(*response_headers) == code) {
```

Comment:
Should we be modeling this as an `unordered_set`? I'm guessing most of these time this list will be only 1 or 2 items long, so it's a bit of a wash, but logically it's the behavior we're describing here.

Reply:
I've been told elsewhere to use std::vector for small collections but I don't really have a strong opinion here. Vector might be slightly faster but a set makes more sense conceptually. Happy to go either way

Reply:
I'll keep it a `vector`